gr-blocks/lib/file_source_impl.cc

```diff
@@ -105,45 +89,37 @@ namespace gr {
         return 0;
       }
 
-      if ((seek_point < (long)d_start_offset_items)
-          || (seek_point > (long)(d_start_offset_items+d_length_items-1))) {
+      if ((seek_point < d_start_offset_items)
+          || (seek_point > (d_start_offset_items+d_length_items-1))) {
         GR_LOG_WARN(d_logger, "bad seek point");
         return 0;
       }
 
-      return fseek((FILE*)d_fp, seek_point * d_itemsize, SEEK_SET) == 0;
+      return fseeko((FILE*)d_fp, seek_point * d_itemsize, SEEK_SET) == 0;
     }
 
 
     void
     file_source_impl::open(const char *filename, bool repeat,
-                           size_t start_offset_items, size_t length_items)
+                           uint64_t start_offset_items, uint64_t length_items)
     {
       // obtain exclusive access for duration of this function
       gr::thread::scoped_lock lock(fp_mutex);
 
-      int fd;
-
-      // we use "open" to use to the O_LARGEFILE flag
-      if((fd = ::open(filename, O_RDONLY | OUR_O_LARGEFILE | OUR_O_BINARY)) < 0) {
```

Comment:
Why did this go and `fopen` is used below instead of `fdopen`? Probably it needs some smarter checking for non-existence.

Reply:
Well you actually pointed that out in the same moment.

Reply:
The reason ::open was being used was to open with O_LARGEFILE. Now that FILE_OFFSET_BITS=64 is being used, the d_new_fp = fopen() is all that's required.

This fix is to just remove the fclose(d_new_fp).

Reply:
Ah, right the throw is just after the fclose()